src/pyj/select.pyj

```diff
@@ -53,14 +53,12 @@ def first_visible_word():
             if r?:
                 return r
 
-
 def span_id_at_point(x, y):
     elements = document.elementsFromPoint(x, y)
     for element in elements:
-        if element.tagName.toUpperCase() is 'SPAN' and element.id:
-            return element.id
-    return None
-
+        spans = element.querySelectorAll('span[id]')
```

Comment:
This will not work when element is itself a span querySelectorAll only matches descendants. Also if you are just going to use the first result from querySelectorAll, more efficient to use querySelector()

Reply:
You will need to pass in a list of all SMIL ids to the iframe when the html file is loaded. Then look for the first visible element that has one of these ids being careful to exclude body.  And thinking about it some more this element could be a large paragraph o ftext, for example, whith some of it on the previous page and some on screen. So you would also need to scroll the element into view when starting playback so that its start is onscreen as the audio starts.

Reply:
I tried that method. If the <p> have padding and the consequent first_id iterate through the range of coordinate, it will click on the padding and miss the span element entirely.

Reply:
OK, in any case see is_anchor_on_screen(). You can just iterate over the list of anchors to get those on screen, and pck the one that is highest up.